vtex/loaders/intelligentSearch/productListingPage.ts

```diff
@@ -202,7 +202,11 @@ const queryFromPathname = (
 ) => {
   const pathList = path.split("/").slice(1);
 
-  const isPage = Boolean(pageTypes.length);
+  const isPage = pageTypes.length > 1 ||
+    (pageTypes.length == 1 &&
+      pageTypes[0].pageType !== "FullText" &&
+      pageTypes[0].pageType !== "Category" &&
+      pageTypes[0].pageType !== "SubCategory");
```

Comment:
it is not clear to me why it should have length exactly 1. Can you extract in a variable and put a comment on top of it?

Also: I think it could be more readable if you extract the possibilities to a const and then use `includes`

like

`["FullText", "Category", "SubCategory"].includes(pageType)`
```suggestion
  const [firstPageThatYouShouldPutAgoodNameWhyItShouldBeTheFirstParameter, ...pageTypes] = pageTypes;
  const isPage = pageTypes.length > 0 || ["FullText", "Category", "SubCategory"].includes(firstPageThatYouShouldPutAgoodNameWhyItShouldBeTheFirstParameter);
```